sdk/lib/opentelemetry/sdk/trace/samplers.rb

```diff
@@ -85,6 +88,25 @@ def self.trace_id_ratio_based(ratio)
 
           TraceIdRatioBased.new(ratio)
         end
+
+        # Returns a new sampler.
+        #
+        # @param [Numeric] ratio The desired sampling ratio.
+        #   Must be within [0.0, 1.0].
+        # @raise [ArgumentError] if ratio is out of range
+        def self.consistent_probability_based(ratio)
+          raise ArgumentError, 'ratio must be in range [0.0, 1.0]' unless (0.0..1.0).include?(ratio)
+
+          ConsistentProbabilityBased.new(ratio)
+        end
+
+        # Returns a new sampler.
+        #
+        # @param [Sampler] root The sampler to which the sampling
+        #   decision is delegated for spans with no parent (root spans).
+        def self.parent_consistent_probability_based(root:)
+          ParentConsistentProbabilityBased.new(root)
```

Comment:
Should we raise `ArgumentError` if you don't pass a root span sampler?

From the spec:
> The ParentConsistentProbabilityBased Sampler constructor SHOULD take a single Sampler argument, which is the Sampler to use in case the ParentConsistentProbabilityBased Sampler is called for a root span.

So I certainly understand why we might not want to enforce the argument, but as the code stands now it'll just blow up at runtime (which is definitely not what we want). Alternately we could address the `nil` root-sampler case directly in `ParentConsistentProbabilityBased` and just ... idk, always sample or something?

Reply:
The code matches what we do with the required `root` parameter to `parent_based`: https://github.com/open-telemetry/opentelemetry-ruby/blob/6bec625ef49004f364457c26263df421526b60d6/sdk/lib/opentelemetry/sdk/trace/samplers.rb#L67-L73

Maybe I'm holding it wrong, but I think the required named parameter is the right approach here. To pass `nil`, someone would have to explicitly configure it as `parent_consistent_probability_based(root: nil)` which seems like "if you do this, you deserve the 💥 ". You're not going to end up with this by accident.